Sources/Request/Request/RequestParams/QueryParam.swift

```diff
@@ -9,34 +9,29 @@ import Foundation
 
 /// A key-value pair for a part of the query string
 public struct QueryParam: RequestParam {
-    public var type: RequestParamType = .query
-    public var key: String?
-    public var value: Any?
+    private var key: String
+    private var value: Any?
     
-    public init(_ key: String, value: String) {
+    public init(_ key: String, value: Any?) {
```

Comment:
I think this should still only support `String`, as we don't know how they want it to be encoded into the URL.